neutone_sdk/parameter.py

```diff
@@ -63,8 +68,63 @@ def __init__(
             description,
             default_value,
             used,
-            NeutoneParameterType.KNOB,
+            NeutoneParameterType.CONTINUOUS,
+        )
+        assert (
+            0.0 <= default_value <= 1.0
+        ), "`default_value` for continuous params must be between 0 and 1"
+
+
+class CategoricalNeutoneParameter(NeutoneParameter):
+    """
+    Defines a categorical Neutone Parameter that the user can use to control a model.
+
+    The name and the description of the parameter will be shown as a tooltip
+    within the UI.
+    `n_values` must be an int greater than or equal to 2 and less than or equal to
+    `constants.MAX_N_CATEGORICAL_VALUES`.
+    `default_value` must be in the range [0, `n_values` - 1].
+    `labels` is a list of strings that will be used as the labels for the parameter.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        description: str,
+        n_values: int,
+        default_value: int,
+        labels: Optional[List[str]] = None,
+        used: bool = True,
+    ):
+        super().__init__(
+            name, description, default_value, used, NeutoneParameterType.CATEGORICAL
+        )
+        assert 2 <= n_values <= constants.MAX_N_CATEGORICAL_VALUES, (
+            f"`n_values` for categorical params must between 2 and "
+            f"{constants.MAX_N_CATEGORICAL_VALUES}"
         )
+        assert (
+            0 <= default_value <= n_values - 1
+        ), "`default_value` for categorical params must be between 0 and `n_values`-1"
+        self.n_values = n_values
+        if labels is None:
+            labels = [str(idx) for idx in range(n_values)]
+        else:
+            assert len(labels) == self.n_values, "labels must have `n_values` elements"
+        assert all(
+            len(label) < constants.MAX_N_CATEGORICAL_LABEL_CHARS for label in labels
+        ), (
+            f"All labels must have length less than "
+            f"{constants.MAX_N_CATEGORICAL_LABEL_CHARS} characters"
+        )
+        self.labels = labels
+
+    def to_metadata_dict(self) -> Dict[str, str]:
+        """Returns a string dictionary containing the metadata of the parameter."""
+        data = super().to_metadata_dict()
+        data["n_values"] = str(self.n_values)
+        data["labels"] = "\t".join(self.labels)
```

Comment:
Why is this separated by \t?

Reply:
Just a habit from always preferring to use TSV over CSV lol
If we do the JSON dumping to a field approach I mentioned in the previous PR's comment then this will become a non-issue.